crates/nu-command/src/path/join.rs

```diff
@@ -25,6 +32,20 @@ impl Command for SubCommand {
         "path join"
     }
 
+    #[cfg(windows)]
+    fn signature(&self) -> Signature {
+        Signature::build("path join")
+            .input_output_types(vec![
+                (Type::String, Type::String),
+                (Type::List(Box::new(Type::String)), Type::String),
+                (Type::Record(vec![]), Type::String),
+                (Type::Table(vec![]), Type::List(Box::new(Type::String))),
+            ])
+            .switch("escape", "use double backslash instead of one.", None)
+            .rest("append", SyntaxShape::String, "Path to append to the input")
+    }
+
+    #[cfg(not(windows))]
```

Comment:
Is it really necessary to make this a platform specific signature?

As devil's advocate is a `\ ` escape of spaces in a unix string also a `--escaped` path to join?

This leads me to question if this is the right place to make hard platform differences? `path join` inherently has a path separator platform specificity but paths are in principle just a array of bytes that have a meaning on the operating system based on the path separator that is OS specific.

Escaping is something that has a meaning both in literals and when passing quoted or unquoted strrings as arguments to external programs/protocols through a string API. Not sure if this really applies correctly to `path join`